### ISSUE TITLE
compute_volume_attach docs use older "element()" method instead of "[index]" syntax

### ISSUE THREAD

**Issue by elpetak** [terraform-providers/terraform-provider-openstack#406]
Hi there,

On this commit: 
https://github.com/terraform-providers/terraform-provider-openstack/commit/b3276c8f7eb9d369f507e9385518d09ae8b50ba3#diff-84caf3d453caa7767d787987ff893a9d

Docs where updated to explain that using count and element() for volume creation and attachment didn't ensure ordering.

There are other issues and commits (a lot of them) about how using count and element() on volume attachment makes the attachments get destroyed/recreated when the count is increased.

It took me a while to find it but it all comes down to this comment here:
https://github.com/hashicorp/terraform/issues/3449#issuecomment-299335806

Bottom line is, using the [count.index] method instead of element() works a lot better.
Using the [] method now I can increase the volume count and new vols are created and attached without affecting existing ones. 

I think the docs should be updated to include this newer method for referencing elements in a list since it has many advantages.

Thanks.

**Comment by jtopjian** [comment]
@elpetak Thanks for the feedback.

If you'd like to submit a PR, I'd be happy to review and merge it.
